File: longest-substring-without-repeating.cs
public class Solution {
    
	public int LengthOfLongestSubstring(string s) {

			if (s.Length == 0) {
				return 0;
			}

			if (s.Length == 1) {
				return 1;
			}

			int currentBiggest = 0;
			int biggest = 0;
			bool biggestWasSet = false;
			Dictionary<char, int> chars = new Dictionary<char, int>();	
			for (int i = 0; i < s.Length; i++) {
				if (biggestWasSet) {
					chars = new Dictionary<char,int>();
					biggestWasSet = false;
				}

				for (int k = i; k < s.Length; k++) {
					bool keyExists = chars.ContainsKey(s[k]);

					if (keyExists) {
						if (biggest < currentBiggest){ 
							biggest = currentBiggest;
						}
						biggestWasSet = true;
						currentBiggest = 0;
						break;
					}
					else {
						currentBiggest++;
						chars.Add(s[k], 1);
					}

				}
			}    
			
			return biggest;
		}
}

File: longest-substring-without-repeating.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) 
    {
        int biggest = 1;
        int leftPointer = 0;
        int rightPointer = 1;
        int counter = 1;

        std::map<char, int> characterMap;

        if (s.size() == 0)
        {
            return 0;
        }

        if (s.size() == 1)
        {
            return 1;
        }

        characterMap.insert({s[leftPointer], 1});

        while (rightPointer < s.size())
        {
            if (characterMap.find(s[rightPointer]) != characterMap.end())
            {
                // the character already exists, remove one from the left pointer

                characterMap.erase(s[leftPointer]);

                leftPointer++;

                counter--;

                if (leftPointer == rightPointer)
                {
                    characterMap.insert({s[leftPointer], 1});
                    rightPointer++;
                    counter = 1;
                }
            }
            else
            {
                characterMap.insert({s[rightPointer], 1});
                rightPointer++;
                counter++;

                if (counter > biggest)
                {
                    biggest = counter;
                }
            }
        }

        return biggest;    
    }
};
